KAA-1113: Changed content and structure of documentation style guide.

hunch7 opened the pull request:
Examples will be added upon review results.
Contribution guide is still in progress and will probably be on a separate page.

Commit:
KAA-1113: Changed content and structure of documentation style guide.
Examples will be added upon review results.
Contribution guide is still in progress and will probably be on a separate page.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,170 @@ sort_idx: 30
+
+The guide contains instructions on how to write and contribute documentation content.
+
+The target audience of this guide is Kaa development team.
```
s/Kaa development team/anyone contributing to Kaa documentation

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,170 @@ sort_idx: 30
+
+The target audience of this guide is Kaa development team.
+
+The purpose of this document is to achieve as much consistency as possible in terms of format, style, content, and structure of the Kaa project documentation.
```
Too many paragraphs, if you ask me.

akokhanovskyi replied:
Can you merge some?

hunch7 replied:
Ok, fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,170 @@ sort_idx: 30
+* TOC represents a list of the page headings and subheadings.
+If your page contains only one heading, TOC is not required.
+* Right below the TOC, include a short introduction explaining the purpose of the page and what the reader can learn from it.
+* The introduction should be 1-10 sentences long, but not more than 5 lines of text.
```
(rule of thumb)

akokhanovskyi replied:
s/The introduction should be/As a rule of thumb, the introduction should be

hunch7 replied:
Done.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,170 @@ sort_idx: 30
+* Keep your titles and headings short and precise, 1-5 words will do.
+* Keep sentence structure simple.
+Try to limit sentences to one clause.
+Individuals who have language difficulties, non-native English speakers, and some people who are deaf or hard of hearing may have difficulty understanding longer, complicated sentences.
```
Deaf? Are we talking machine narration here?

hunch7 replied:
Yes. Accessibility is the principle here.

akokhanovskyi replied:
Okay

hunch7 committed:
Update index.md

Merged two lines in the introduction part.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+Individuals who have language difficulties, non-native English speakers, and some people who are deaf or hard of hearing may have difficulty understanding longer, complicated sentences.
+* If you have sentences longer than 25 words, try to break them up or condense them.
+If you can’t, make sure they are in plain English.
+* Keep paragraphs short or create small sections or text groupings.
```
Add a rule of thumb for the paragraph length here.

hunch7 replied:
Done.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Put key information near the beginning of the content.
+* Provide descriptions that do not require pictures, or provide both pictures and written descriptions.
+To test whether the writing is effective, try removing, one at a time, first the words and then the pictures.
+With only one method, can you still figure out what to do?
```
"Method" of..? Clarify

hunch7 replied:
Fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Minimize the number of steps in a procedure.
+Individuals who have cognitive impairments may have difficulty following procedures that have many steps.
+Keep the steps simple, and keep the user oriented.
+* Provide links to related content within the documentation to help readers get more information.
```
Also link to auto-generated docs. (API docs, etc)

hunch7 replied:
Fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
-* Avoid using please in instructions unless some step in the procedure causes inconvenience for the user or represents a workaround for some system limitation.
-
-## Headings and capitalization
+* Use short, plain words as much as possible.
```
Merge with line 57?

hunch7 replied:
Ok, fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+The overuse of exclamation points can make content seem effusive or ingratiating.
+Save exclamation points for when they count.
+* Be mindful of globalization and cultural considerations before you use colloquialisms or idioms.
+* Do not use non-English words or phrases, such as _de facto_ or _ad hoc_, in English content, even if you think these terms are generally known and understood.
```
How about etc., eg., ie., and so on?

hunch7 replied:
I propose that we have only two exceptions for this rule: etc. and e.g. They are used very often and their English equivalents are too long. I'll put this in my next commit.

akokhanovskyi replied:
I use "i.e." a lot too :)

hunch7 replied:
No problem, but not in our documentation, please :)

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Provide links from the main text to the glossary.
+* Do not create a new term if a term describing a concept already exists in the glossary.
+* If you need to create a new term, verify that the term that you select is not already in use to mean something else.
+
```
When using defined terms, make sure you use them exactly as defined. Shortcutting only works when it's defined. Examples: "Administrative UI" is not equal to "Admin UI" unless defined so; "tenant administrator" is not the same as "tenant", etc.

hunch7 replied:
Fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+## Grammar
+* Give preference to present tense over past and future tenses.
+* Give preference to simple verb forms over perfect and continuous forms.
+* Put the action of the sentence in the verb, not in the nouns.
```
At least this item could use an example for clarity.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+Either supply a direct object or use an intransitive verb instead.
+* Avoid using weak, vague verbs such as _be_, _have_, _make_, and _do_.
+Such verbs are sometimes necessary, but use a more descriptive verb whenever you can.
+* Use imperative mood for procedures, formulate your instructions as giving commands to the user.
```
+example

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Avoid using weak, vague verbs such as _be_, _have_, _make_, and _do_.
+Such verbs are sometimes necessary, but use a more descriptive verb whenever you can.
+* Use imperative mood for procedures, formulate your instructions as giving commands to the user.
+* Use indicative mood for conveying information such as facts, questions, assertions, or explanations.
```
+example

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Put the action of the sentence in the verb, not in the nouns.
+* Choose single-word verbs over multiple-word verbs.
+* Don’t convert verbs to nouns and nouns to verbs.
+* Do not use a transitive verb without a direct object.
```
+example

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+Second person supports a friendly tone because it connects you with the user.
+* Use first person sparingly.
+First person is appropriate when writing from the point of view of the user.
+* Avoid using more than 3 nouns in a noun string as it may be confusing or ambiguous even for native English speakers.
```
+example? :)

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+  * Names of commands, attributes, constants, methods, fields, predefined classes, databases, events, UI elements.
+  * User input.
+* Use _italic_ for:
+  * Technical terms on first mention.
```
Personally, I think italics does not stand out in the page well enough (as the term definitions deserve). Why don't we use **_bold italics_** for this?

hunch7 replied:
I don't think it's a good idea, because there will be too much bold text distracting user's attention. From my experience, users do not see the difference between bold and bold italics. Also, too much bold text messes up with headings and subheadings, which looks ugly and irritates users :)

akokhanovskyi replied:
I tend to agree that there is too much bold. I would reduce the amount of bold stuff in favor of term definitions. These are important. There is no good reason, for example, why user input should be bold.
Would I open a page to quickly find user input? No.
Would I open a page to quickly find key terms? Yes.

hunch7 replied:
Agreed. I will analyze and fix this.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+  * Technical terms on first mention.
+  * Emphasis.
+  * Placeholders and parameter names.
+* Use links for names of pages and sections.
```
I'm not sure what this means.

hunch7 replied:
I define link as one of the formatting elements. So, when we refer to other sections of Kaa documentation, or external web pages, we should put them in the text as links.

akokhanovskyi replied:
ok, can you rephrase to clarify?

hunch7 replied:
Done.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
-* Format all JSON files with this [online tool](https://jsonformatter.curiousconcept.com/), "4 space tab" profile.
+* Do not use capitalization for no apparent reason. Use lowercase unless uppercase is justified.
+* Use regular text (capitalized where applicable) for all other purposes, such as:
+  * Names of user-defined classes, folders, files and file extensions.
```
This falls into code formatting section, doesn't it?

hunch7 replied:
This is for occasional inline references that are not code examples. E.g.: Open the Applications folder.

akokhanovskyi replied:
I'm not clear what "user-defined classes" are doing in this list. Classes are code - no?

hunch7 replied:
Ok, I removed classes from this guideline.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Introduce a table with a sentence that ends with a period, not a fragment that ends with a colon.
+* If a table is titled, an introductory sentence does not have to immediately precede the table.
+* Put the table title above the table.
+* Table dimensions must be visible on a minimum screen resolution, typically 800 pixels by 600 pixels.
```
s/dimensions//

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* If a table is titled, an introductory sentence does not have to immediately precede the table.
+* Put the table title above the table.
+* Table dimensions must be visible on a minimum screen resolution, typically 800 pixels by 600 pixels.
+* Put footnote explanations at the end of the table, not at the bottom of the page.
```
This is only relevant to printed pages. We are talking online documentation here.

hunch7 replied:
No, it's not about printed pages. This one means that you should stick to logical flow of the content and put explanations below the table to keep users focused. Putting some other info after a table and then going back to the table explanation is not effective.

akokhanovskyi replied:
Okay, I was thinking about long tables with page breaks.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Put the table title above the table.
+* Table dimensions must be visible on a minimum screen resolution, typically 800 pixels by 600 pixels.
+* Put footnote explanations at the end of the table, not at the bottom of the page.
+* Tables can be used to simulate frames.
```
I doubt this is relevant

hunch7 replied:
Ok, fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+
+## Notes, tips, and cautions
+* Use notes sparingly so that they remain effective in drawing the user’s attention.
+* In general, try to use only one note in a Help topic.
```
What is "Help topic"?

hunch7 replied:
I'll put it this way: 'Try using not more than one note per subheading.'

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Use _important note_ with the heading “Important” to provide information that is essential to the completion of a task.
+Users can disregard information in a note and still complete a task, but they should not disregard an important note.
+* Use _caution_ with the heading “Caution” to advise users that failure to take or avoid a specific action could result in loss of data, damage of hardware, and other issues of similar importance.
+
```
You forgot to define the note formatting.

hunch7 replied:
Yes, I fixed this, but partially. To finalize it, I need to get the corresponding CSS classes from the Design team.

akokhanovskyi replied:
Ah, so we we're making custom note styles? Good.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+* Use _caution_ with the heading “Caution” to advise users that failure to take or avoid a specific action could result in loss of data, damage of hardware, and other issues of similar importance.
+
+## Code examples
+* Format all JSON files with this [online tool](https://jsonformatter.curiousconcept.com/) using the "4 space tab" profile.
```
JSON is not so important to be the first item in the list.

hunch7 replied:
Fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
-Rather than explaining the same topic multiple times in multiple locations, modularize and use links to fragment identifiers when applicable.
-
+See [Jekyll formatting](#jekyll-formatting).
+* Enable the syntax highlight for the language of the code example whenever available.
```
How? Be specific.

hunch7 replied:
Fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -11,62 +11,168 @@ sort_idx: 30
+  * Source file names.
+  * Numbers when they are used as code.
 
 ## Markdown formatting
```
I would actually move this to the very beginning of the page

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -212,8 +319,8 @@ Eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dic
-Kaa documentation is a part of [Kaa source code](https://github.com/kaaproject/kaa) - it can be found in the `doc/` folder.
-A full description of contributing to Kaa project can be found at [How to contribute]({{root_url}}Customization-guide/How-to-contribute/).
+Kaa documentation is a part of [Kaa source code](https://github.com/kaaproject/kaa) and is located in the doc/ folder.
+For full description of the contributing process, see [How to contribute]({{root_url}}Customization-guide/How-to-contribute/).
```
s/contributing/contribution

hunch7 replied:
Fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -230,7 +337,7 @@ Key differences between contributing source code and documentation are:
   * [Rouge](https://github.com/jneen/rouge)
 
-   Detailed installation instructions for selected platforms:
+   See the detailed installation instructions in the table below. Click the appropriate tab to select your platform:
```
"selected platforms" used to mean that not every world's platform is covered.

hunch7 replied:
Ok, I understand. But I don't think we should mention that :) My edit was towards making it a call-to-action description. I'll rephrase the 2nd sentence this way: 'Click the tab to select a platform'.

hunch7 committed:
Update index.md

Guidelines updated. Design team will provide collapsible blocks for the examples.

hunch7 committed:
Update index.md

Rewrote and rearranged some sections.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -105,10 +62,235 @@ Rather than explaining the same topic multiple times in multiple locations, modu
+* Use **bold** for:
+  * Technical terms that are not in the Glossary, on their first mention.
+  * Titles of windows and dialog boxes.
+  * Names of UI elements, commands, attributes, constants, methods, fields, predefined classes, databases, events.
```
I'd rather make these italic

hunch7 replied:
As discussed, we leave these in bold. Also, users often seek for particular instructions on how to do something, so bold makes sense here.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -105,10 +62,235 @@ Rather than explaining the same topic multiple times in multiple locations, modu
+  * Technical terms that are not in the Glossary, on their first mention.
+  * Titles of windows and dialog boxes.
+  * Names of UI elements, commands, attributes, constants, methods, fields, predefined classes, databases, events.
+  * User input.
```
Make italic

hunch7 replied:
As discussed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -105,10 +62,235 @@ Rather than explaining the same topic multiple times in multiple locations, modu
+* Start every list either with a heading or with a sentence followed by a colon.
+* Capitalize the first word of each list entry.
+* End each list entry with a period.
+* Do not use ending punctuation for lists where all entries are three words or fewer.
```
Merge with the previous line: ", however, do not use..."

hunch7 replied:
Ok, fixed.

akokhanovskyi commented on doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md:
```diff
@@ -105,10 +62,235 @@ Rather than explaining the same topic multiple times in multiple locations, modu
+## Use of terminology
+* Whenever possible, use common English instead of using terminology.
+Use terminology when you need to be particular and precise about something.
+* Provide term definition on its first mention.
```
Add: "Link to the Glossary for terms that are in the Glossary. For others, use **bold** formatting on the first mention in a page."

I know this is also defined in other places. That's fine.

hunch7 replied:
Ok, fixed.

hunch7 committed:
Update index.md

Minor rephrasing.

Commit:
Typos fixed, one redundant line deleted.
